featurebyte/models/use_case.py

```diff
@@ -0,0 +1,60 @@
+"""
+Use Case model
+"""
+from typing import List, Optional
+
+import pymongo
+from pydantic import Field
+
+from featurebyte.models.base import (
+    FeatureByteCatalogBaseDocumentModel,
+    PydanticObjectId,
+    UniqueConstraintResolutionSignature,
+    UniqueValuesConstraint,
+)
+
+
+class UseCaseModel(FeatureByteCatalogBaseDocumentModel):
+    """
+    UseCaseModel represents a use case within a feature store
+
+    context_id: PydanticObjectId
+        The context id of ContextModel
+    target_id: PydanticObjectId
+        The target id of TargetModel
+    observation_table_ids:  List[PydanticObjectId]
+        The observation table ids of ObservationTableModel
+    default_preview_table_id: PydanticObjectId
+        The default preview table from observation_table_ids
+    default_eda_table_id: PydanticObjectId
+        The default eda table id of from observation_table_ids
+    """
+
+    context_id: PydanticObjectId
+    target_id: PydanticObjectId
+    observation_table_ids: List[PydanticObjectId] = Field(default=[])
```

Comment:
Should we use `default_factory=list` here?

Reply:
Seems like `default` should work also (https://github.com/pydantic/pydantic/blob/5a1fdfad1f1e5108c11c76a451f9a0e671bef8e7/pydantic/fields.py#L501). Previously, i worry that it could copy by reference by not copy by value.